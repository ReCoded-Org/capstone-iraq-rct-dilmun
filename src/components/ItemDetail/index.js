export { default } from './ItemDetail'
export { default as SlideShow } from './SlideShow'
