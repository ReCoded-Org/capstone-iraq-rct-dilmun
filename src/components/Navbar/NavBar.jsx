import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { NavLink, useHistory } from 'react-router-dom'
import i18n from 'i18next'
import { useSelector, useDispatch } from 'react-redux'
import firebase from '../../firebase'
import Logo from '../../logo.svg'
import {
  HOME_ROUTE,
  PROFILE_ROUTE,
  CONTACT_ROUTE,
  ABOUT_ROUTE,
} from '../../router'
import Login from '../LogIn'
import userIcon from '../../assets/userIcon.png'
import { lOGOUT } from '../../redux/Authentication/ActionTypes'
import { LogOut } from '../../redux/Authentication/AuthenticationActions'

export default function Navbra() {
  const dispatch = useDispatch()
  const user = useSelector(state => state.authentication)

  const history = useHistory()
  const [profileDropDown, setprofileDropDown] = useState(false)
  const [navCollapse, setnavCollapse] = useState(false)

  const { t } = useTranslation()
  const handleLanguageChange = lang => {
    i18n.changeLanguage(lang)
  }

  const signOut = () => {
    firebase.auth().signOut()
    setprofileDropDown(false)
    dispatch(LogOut(lOGOUT))
    localStorage.removeItem('loggedInUser')
    history.push('/')
  }
  return (
    <>
      <nav className="bg-white z-50 sticky top-0  ">
        <div className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
          <div className="relative flex items-center justify-between h-16">
            <div className="absolute inset-y-0 left-0 flex items-center sm:hidden">
              <button
                type="button"
                className="inline-flex items-center justify-center p-2 rounded-md text-blue hover:text-darkBlue  focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                aria-controls="mobile-menu"
                aria-expanded="false"
                onClick={() => setnavCollapse(!navCollapse)}
              >
                <span className="sr-only">Open main menu</span>

                <svg
                  className="block h-6 w-6"
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke="currentColor"
                  aria-hidden="true"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    d="M4 6h16M4 12h16M4 18h16"
                  />
                </svg>

                <svg
                  className="hidden h-6 w-6"
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke="currentColor"
                  aria-hidden="true"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    d="M6 18L18 6M6 6l12 12"
                  />
                </svg>
              </button>
            </div>
            <div className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start">
              <div className="flex-shrink-0 flex items-center">
                <img
                  className="block lg:hidden h-8 w-auto"
                  src={Logo}
                  alt="logo"
                />
                <img
                  className="hidden lg:block h-8 w-auto"
                  src={Logo}
                  alt="logo"
                />
              </div>
              <div className="hidden sm:block sm:ml-6">
                <div className="flex space-x-4 text-blue">
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    exact
                    to={HOME_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('navbar.home')}
                  </NavLink>
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    to={CONTACT_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('navbar.contact')}
                  </NavLink>
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    to={ABOUT_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('navbar.about')}
                  </NavLink>
                  <NavLink
                    className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                    to={PROFILE_ROUTE}
                    activeClassName="text-darkBlue font-bold"
                  >
                    {t('footer.profile')}
                  </NavLink>
                </div>
              </div>
            </div>
            <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
              <button
                onClick={() => handleLanguageChange(t('language'))}
                type="button"
                className=" hover:bg-darkBlue hover:text-white items-center px-3   text-blue  rounded inline-flex focus:outline-none items-center border border-solid border-blue justify-center"
              >
                <i className="fas fa-globe pr-1" />
                <div className="text-center">{t('langText')}</div>
              </button>

              <div className="ml-3 relative">
                <div>
                  {user.isLoggedIn ? (
                    <button
                      onClick={() => setprofileDropDown(!profileDropDown)}
                      type="button"
                      className="bg-gray-800 flex text-sm rounded-full focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                      id="user-menu"
                      aria-expanded="false"
                      aria-haspopup="true"
                    >
                      <span className="sr-only">Open user menu</span>

                      <img
                        className="h-8 w-8 rounded-full"
                        src={user ? user.user.photo : userIcon}
                        alt=""
                      />
                    </button>
                  ) : (
                    <Login />
                  )}
                </div>

                <div
                  className={`${
                    profileDropDown ? ' show' : ' hidden'
                  } origin-top-right absolute right-0 border border-t-0 boder-grey mt-2 w-48 rounded-md items-center shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none`}
                  role="menu"
                  aria-orientation="vertical"
                  aria-labelledby="user-menu"
                >
                  <NavLink
                    to={PROFILE_ROUTE}
                    className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white"
                    role="menuitem"
                  >
                    {user.isLoggedIn ? user.user.name : ''}
                  </NavLink>
                  <NavLink
                    to="/"
                    className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white "
                    role="menuitem"
                  >
                    Settings
                  </NavLink>
                  <button
                    type="button"
                    onClick={signOut}
                    className="m-auto text-left block focus:outline-none px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white w-full"
                    role="menuitem"
                  >
                    Sign out
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>

        <div
          className={`${navCollapse ? 'show' : 'hidden'} sm:hidden`}
          id="mobile-menu"
        >
          <div className="px-2 pt-2 pb-3 space-y-1 text-blue">
            <NavLink
              to={HOME_ROUTE}
              className="text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
            >
              Home
            </NavLink>
            <NavLink
              to={CONTACT_ROUTE}
              className="text-blue  hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
            >
              Contact
            </NavLink>
            <NavLink
              to={ABOUT_ROUTE}
              className="text-blue hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
            >
              About
            </NavLink>
            <NavLink
              to={PROFILE_ROUTE}
              className="hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
              activeClassName="text-darkBlue font-bold"
            >
              Profile
            </NavLink>
          </div>
        </div>
      </nav>
    </>
  )
}
