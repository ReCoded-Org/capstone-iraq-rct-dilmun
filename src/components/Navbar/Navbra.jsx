import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import i18n from 'i18next'
import Logo from '../../logo.svg'

export default function Navbra() {
    const [profileDropDown, setprofileDropDown] = useState(false)
    const [navCollapse, setnavCollapse] = useState(false)
    const [isLogedIn, setIsLogedIn] = useState(true)
    const { t } = useTranslation()
    const handleLanguageChange = (lang) => {
        i18n.changeLanguage(lang)
    }
    return (
        <nav className="bg-white">
            <div className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
                <div className="relative flex items-center justify-between h-16">
                    <div className="absolute inset-y-0 left-0 flex items-center sm:hidden">
                        <button
                            type="button"
                            className="inline-flex items-center justify-center p-2 rounded-md text-blue hover:text-darkBlue  focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                            aria-controls="mobile-menu"
                            aria-expanded="false"
                            onClick={() => setnavCollapse(!navCollapse)}
                        >
                            <span className="sr-only">Open main menu</span>

                            <svg
                                className="block h-6 w-6"
                                xmlns="http://www.w3.org/2000/svg"
                                fill="none"
                                viewBox="0 0 24 24"
                                stroke="currentColor"
                                aria-hidden="true"
                            >
                                <path
                                    strokeLinecap="round"
                                    strokeLinejoin="round"
                                    strokeWidth="2"
                                    d="M4 6h16M4 12h16M4 18h16"
                                />
                            </svg>

                            <svg
                                className="hidden h-6 w-6"
                                xmlns="http://www.w3.org/2000/svg"
                                fill="none"
                                viewBox="0 0 24 24"
                                stroke="currentColor"
                                aria-hidden="true"
                            >
                                <path
                                    strokeLinecap="round"
                                    strokeLinejoin="round"
                                    strokeWidth="2"
                                    d="M6 18L18 6M6 6l12 12"
                                />
                            </svg>
                        </button>
                    </div>
                    <div className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start">
                        <div className="flex-shrink-0 flex items-center">
                            <img
                                className="block lg:hidden h-8 w-auto"
                                src={Logo}
                                alt="logo"
                            />
                            <img
                                className="hidden lg:block h-8 w-auto"
                                src={Logo}
                                alt="logo"
                            />
                        </div>
                        <div className="hidden sm:block sm:ml-6">
                            <div className="flex space-x-4">
                                <a
                                    href="/"
                                    className=" text-darkBlue hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                                >
                                    {t('navbar.home')}
                                </a>
                                <a
                                    href="/"
                                    className="text-blue hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                                >
                                    {t('navbar.contact')}
                                </a>
                                <a
                                    href="/"
                                    className="text-blue  hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
                                >
                                    {t('navbar.about')}
                                </a>
                            </div>
                        </div>
                    </div>
                    <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
                        <button
                            onClick={() => handleLanguageChange(t('language'))}
                            type="button"
                            className=" hover:bg-darkBlue hover:text-white items-center px-3   text-blue  rounded inline-flex focus:outline-none items-center border border-solid border-blue justify-center"
                        >
                            <i className="fas fa-globe pr-1" />
                            <div className="text-center">{t('langText')}</div>
                        </button>

                        <div className="ml-3 relative">
                            <div>
                                {isLogedIn ? (
                                    <button
                                        onClick={() =>
                                            setprofileDropDown(!profileDropDown)
                                        }
                                        type="button"
                                        className="bg-gray-800 flex text-sm rounded-full focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                                        id="user-menu"
                                        aria-expanded="false"
                                        aria-haspopup="true"
                                    >
                                        <span className="sr-only">
                                            Open user menu
                                        </span>

                                        <img
                                            className="h-8 w-8 rounded-full"
                                            src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=htmlFormat&fit=facearea&facepad=2&w=256&h=256&q=80"
                                            alt=""
                                        />
                                    </button>
                                ) : (
                                    <div className="bg-darkBlue  items-center  rounded-full items-center">
                                        <button
                                            type="submit"
                                            className=" py-1 px-3 bg-white  items-center  rounded-full text-darkBlue m-1    focus:outline-none  "
                                        >
                                            Log in
                                        </button>

                                        <button
                                            type="submit"
                                            className=" py-1 px-3   text-white    focus:outline-none  "
                                        >
                                            Signup
                                        </button>
                                    </div>
                                )}
                            </div>

                            <div
                                className={`${
                                    profileDropDown ? ' show' : ' hidden'
                                } origin-top-right absolute right-0 mt-2 w-48 rounded-md items-center shadow-lg py-1 bg-white ring-1 ring-black ring-opacity-5 focus:outline-none`}
                                role="menu"
                                aria-orientation="vertical"
                                aria-labelledby="user-menu"
                            >
                                <a
                                    href="/"
                                    className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white"
                                    role="menuitem"
                                >
                                    Your Profile
                                </a>
                                <a
                                    href="/"
                                    className="block px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white "
                                    role="menuitem"
                                >
                                    Settings
                                </a>
                                <button
                                    type="button"
                                    onClick={() => setIsLogedIn(!isLogedIn)}
                                    className="m-auto block focus:outline-none px-4 py-2 text-sm text-gray-700 hover:bg-darkBlue hover:text-white w-full"
                                    role="menuitem"
                                >
                                    Sign out
                                </button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <div
                className={`${navCollapse ? 'show' : 'hidden'} sm:hidden`}
                id="mobile-menu"
            >
                <div className="px-2 pt-2 pb-3 space-y-1">
                    <a
                        href="/"
                        className="text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
                    >
                        Home
                    </a>
                    <a
                        href="/"
                        className="text-blue  hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
                    >
                        Contact
                    </a>
                    <a
                        href="/"
                        className="text-blue hover:text-darkBlue block px-3 py-2 rounded-md text-base font-medium"
                    >
                        About
                    </a>
                </div>
            </div>
        </nav>
    )
}
