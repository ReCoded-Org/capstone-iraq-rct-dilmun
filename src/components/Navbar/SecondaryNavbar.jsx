import React from 'react'
import uuid from 'react-uuid'
import { NavLink } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

export default function SecondaryNavbar() {
  const { t } = useTranslation()
  const catagories = t('additem.cat', { returnObjects: true })

  const result = Object.keys(catagories).map(key => catagories[key])

  return (
    <ul className="flex overflow-x-auto whitespace-nowrap p-2 justify-items-center border-b border-grey text-center shadow ">
      {result.map(catagory => {
        return (
          <li className="flex-1" key={uuid()}>
            <NavLink
              className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
              to={`/searchresult/${catagory.url}`}
              activeClassName="text-darkBlue"
            >
              {catagory.value}
            </NavLink>
          </li>
        )
      })}
    </ul>
  )
}
