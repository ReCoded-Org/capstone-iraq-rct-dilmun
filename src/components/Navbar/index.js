export { default } from './NavBar'
export { default as SecondaryNavbar } from './SecondaryNavbar'
