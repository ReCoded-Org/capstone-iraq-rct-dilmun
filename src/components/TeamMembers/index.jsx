export { default } from './TeamMembers'
export { default as AboutUs } from './AboutUs'
