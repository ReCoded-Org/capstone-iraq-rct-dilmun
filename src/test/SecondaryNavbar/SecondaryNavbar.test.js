import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import { SecondaryNavbar } from '../../components/Navbar'

it('Secondary Navbar ', () => {
    const tree = renderer
        .create(
            <Router>
                <SecondaryNavbar />
            </Router>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})
