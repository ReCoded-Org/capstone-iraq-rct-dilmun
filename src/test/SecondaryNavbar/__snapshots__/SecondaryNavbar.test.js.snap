// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary Navbar  1`] = `
<ul
  className="flex overflow-x-auto whitespace-nowrap p-2 justify-items-center border-b border-grey text-center shadow "
>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
  <li
    className="flex-1"
  >
    <a
      aria-current={null}
      className=" hover:text-darkBlue px-3 py-2 rounded-md text-sm font-medium"
      href="/undefined"
      onClick={[Function]}
    />
  </li>
</ul>
`;
